content/en/getting_started/tracing/_index.md

```diff
@@ -20,159 +20,201 @@ further_reading:
-pip install ddtrace
-```
+## Test the application
 
-### Create
+Test the application to send traces to Datadog:
 
-On the Ubuntu VM, create the application `hello.py` with the following content:
+1. In a new command prompt, run the following:
 
-```python
-from flask import Flask
-app = Flask(__name__)
+   ```shell
+   curl http://0.0.0.0:5050/
+   ```
+1. Confirm that a random quote is returned.
+   ```text
+   Believe you can and you're halfway there. - Theodore Roosevelt
+   ```
 
-@app.route('/')
-def index():
-    return 'hello world'
+Each time you run the `curl` command, a new trace is sent to Datadog.
 
-if __name__ == '__main__':
-    app.run(host='0.0.0.0', port=5050)
-```
+## Explore traces in Datadog
 
-### Run
+1. In Datadog, go to [**APM** > **Services**][3]. You should see a Python service named `hello`:
 
-Run `hello.py` with `ddtrace` which automatically instruments your application in Datadog:
+   {{< img src="/getting_started/apm/service-catalog.png" alt="Service Catalog shows the new Python service." style="width:100%;" >}}
 
-```shell
-export DD_SERVICE=hello
-ddtrace-run python hello.py
-```
+1. Select a service to view its performance metrics, such as latency, throughput, and error rates.
+1. Go to [**APM** > **Traces**][4]. You should see a trace for the `hello` service:
 
-You should see a similar output to:
+   {{< img src="/getting_started/apm/trace-explorer.png" alt="Trace explorer shows the trace for the hello service." style="width:100%;" >}}
 
-```bash
-* Serving Flask app "hello" (lazy loading)
-  ...
-* Running on http://0.0.0.0:5050/ (Press CTRL+C to quit)
-```
+1. Select a trace to see its details, including the flame graph, which helps identify performance bottlenecks.
 
-### Test
+## Advanced APM setup
 
-Test your application and send your traces to Datadog using `curl`. Your application should be running (as shown above). In a separate command prompt run:
+Up until this point, you let Datadog automatically instrument the `hello.py` application using Single Step Instrumentation. This is great if you want to capture essential traces across common libraries and languages without touching code or manually installing libraries.
 
-```text
-vagrant ssh
-curl http://0.0.0.0:5050/
-```
+However, if you need to collect traces from custom code or require more fine-grained control, you can add [custom instrumentation][6].
 
-This outputs:
+To illustrate this, you are going to import the Datadog Python tracing library into `hello.py` and create a custom span and span tag.
 
-```text
-hello world
-```
+To add custom instrumentation:
 
-After a few minutes, your trace displays in Datadog under the `hello` service. Check the [Service Catalog][14] or [trace list][15].
+1. Install the Datadog tracing library:
 
-{{< img src="getting_started/tracing-services-list.png" alt="Tracing Services List" >}}
+   ```shell
+   pip install ddtrace
+   ```
 
-## Further Reading
+1. Add the highlighted lines to the code in `hello.py`:
```

Comment:
```suggestion
1. Add the highlighted lines to the code in `hello.py` to create a custom span tag `get_quote` and a custom span tag `quote`:
```

Or something like that, just to be a little more explicit about what we're doing here.